### ISSUE TITLE
HttpAuthenticationMechanism SecurityException throwing hangs the REST call

### ISSUE THREAD

**Issue by burmanm** [quarkusio/quarkus#11497]
**Describe the bug**
Using the following example code (resteasy-mutiny and quarkus-security):

```java
// Omitted the rest
import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
import io.quarkus.security.identity.request.AuthenticationRequest;

@ApplicationScoped
public class IdentityAuthMechanism implements HttpAuthenticationMechanism {

    public static final String IDENTITY_HEADER = "x-rh-identity";

    @Override
    public Uni<SecurityIdentity> authenticate(RoutingContext routingContext, IdentityProviderManager identityProviderManager) {
        throw new AuthenticationFailedException();
    }

   // Rest omitted
```

And calling an endpoint secured by it causes the calling http client to hang indefinitely. Easy to test with JAX-RS endpoint like:

```java
    @GET
    @RolesAllowed("read")
    public Multi<Endpoint> getEndpoints(@Context SecurityContext sec) {
        return Multi.createFrom().nothing();
    }
```

And then calling the endpoint. It will never finish to reply anything:

```
[michael@nina notifications-backend]$ curl -v http://localhost:8080/endpoints
*   Trying ::1:8080...
* Connected to localhost (::1) port 8080 (#0)
> GET /api/notifications/v1.0/endpoints HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.71.1
> Accept: */*
> 
^C
```

**Expected behavior**
401 / 403 statusCode.

**Actual behavior**
The REST-client will wait forever (or until timeout) as there's no reply. 

**Environment (please complete the following information):**
 - Output of `java -version`: 11
 - Quarkus version or git rev: 1.7.0.Final

**Comment by jaikiran** [comment]
Hello @burmanm, when this hangs, can you get the `jstack` output of the (Java) process that is running you Quarkus application? 2-3 jstack output separated by around 5 seconds each would help us try and narrow this down quickly.

**Comment by sberyozkin** [comment]
@burmanm Do you get 401 if the JAX-RS resource returns say String ?

**Comment by burmanm** [comment]
First jstack: https://gist.github.com/burmanm/ecf9cd1b386417c87e558901ab051fc3
6 seconds later: https://gist.github.com/burmanm/39b98cbd56da7fc1d9b8465ebb1400a6

**Comment by burmanm** [comment]
@sberyozkin Do you mean using non-reactive-reply?

**Comment by sberyozkin** [comment]
@burmanm Yes, just curious if returning Uni/Multi has some build time expectations about the required response introduced. So yes, using non-reactive reply

**Comment by burmanm** [comment]
No difference.

**Comment by burmanm** [comment]
From another ticket (https://github.com/smallrye/smallrye-mutiny/issues/254), it seems that the Mutiny version works when the error comes from certain methods of Mutiny and the AuthenticationFailedException works. Maybe @cescoffier could share some light to this? So one needs to be careful not to create exceptions when outside the Mutiny-flow.
